src/mcmini_private.cpp

```diff
@@ -547,18 +567,25 @@ mc_search_dpor_branch_with_initial_thread(const tid_t leadingThread)
     mcprintf("Trace %lu, *** DEADLOCK DETECTED ***\n", traceId);
     programState->printTransitionStack();
     programState->printNextTransitions();
+    addResult("*** DEADLOCK DETECTED ***\n");
 
     if (getenv(ENV_FIRST_DEADLOCK) != NULL) {
-      mcprintf("*** Model checking completed! ***\n");
+      printResults();
       mcprintf("Number of transitions: %lu\n", transitionId);
       mc_exit(EXIT_SUCCESS);
     }
   }
 
-  if (programHasNoErrors && getenv(ENV_VERBOSE)) {
-    mcprintf("Trace: %d, *** NO FAILURE DETECTED ***\n", traceId);
-    programState->printTransitionStack();
-    programState->printNextTransitions();
+  static char *verbose = getenv(ENV_VERBOSE);
```

Comment:
Ideally, we'd want to have some sort of configuration flag system that could be queried instead of directly going for `getenv()` (e.g. like that which exists in SimGrid). This is again ok for the moment; a future, more comprehensive change would make this more readable though (e.g. with a C++ class enum)

Reply:
I think you agree that this is easy to read.  Just search on 'ENV_'.  Environment variables are just one more type of global variable.  By using principled naming conventions, grep allows someone to easily find the relevant code.  Can we push this in?